modules/Walking_module/src/WalkingQPInverseKinematics.cpp

```diff
@@ -112,17 +195,97 @@ bool WalkingQPIK::initializeMatrices(const yarp::os::Searchable& config)
         return false;
     }
 
+    // get the regularization term
+    if(!YarpHelper::getiDynTreeVectorDynSizeFromSearchable(config, "joint_regularization",
+                                                           m_regularizationTerm))
+    {
+        yError() << "[initialize] Unable to get the iDynTreeVector from searchable.";
+        return false;
+    }
+
+    iDynTree::toEigen(m_regularizationTerm) = iDynTree::toEigen(m_regularizationTerm) *
+        iDynTree::deg2rad(1);
+
+    if(!iDynTree::parseRotationMatrix(config, "additional_rotation", m_additionalRotation))
+    {
+        yError() << "[initialize] Unable to set the additional rotation.";
+        return false;
+    }
+
+    if(!YarpHelper::getNumberFromSearchable(config, "k_joint_limit_lower_bound", m_kJointLimitsLowerBound))
+    {
+        yError() << "Initialization failed while reading a double.";
+        return false;
+    }
+
+    if(!YarpHelper::getNumberFromSearchable(config, "k_joint_limit_upper_bound", m_kJointLimitsUpperBound))
+    {
+        yError() << "Initialization failed while reading a double.";
+        return false;
+    }
+
     return true;
 }
 
-WalkingQPIK::~WalkingQPIK()
-{ }
+bool WalkingQPIK::initializeHandRetargeting(const yarp::os::Searchable& config)
+{
+    if(!YarpHelper::getNumberFromSearchable(config, "k_posHand", m_kPosHand))
+    {
+        yError() << "Initialization failed while reading k_posHand.";
+        return false;
+    }
+
+    if(!YarpHelper::getNumberFromSearchable(config, "k_attHand", m_kAttHand))
+    {
+        yError() << "Initialization failed while reading k_attHand.";
+        return false;
+    }
+
+    m_handWeightWalkingVector.resize(6);
+    if(!YarpHelper::getYarpVectorFromSearchable(config, "hand_weight_stance",
+                                                m_handWeightWalkingVector))
+    {
+        yError() << "Initialization failed while reading the yarp vector.";
+        return false;
+    }
+
+
+    m_handWeightStanceVector.resize(6);
+    if(!YarpHelper::getYarpVectorFromSearchable(config, "hand_weight_stance",
+                                                m_handWeightStanceVector))
+    {
+        yError() << "Initialization failed while reading the yarp vector.";
```

Comment:
Consider using a slightly different error message to differentiate from the one above.